tests/objcgentest/ObjCGenErrWarnTests.cs

```diff
@@ -0,0 +1,135 @@
+			string tmpDir = Xamarin.Cache.CreateTemporaryDirectory ();
+			var warnerrAssembly = ObjCGeneratorTest.Helpers.Universe.Load ("managedwarn");
+			Assert.NotNull (warnerrAssembly, "warnerrAssembly");
+			var stdout = Console.Out;
+			var stderr = Console.Error;
+
+			try {
+				using (var ms = new MemoryStream ()) {
+					var sw = new StreamWriter (ms);
+
+					Console.SetOut (sw);
+					Console.SetError (sw);
+
+					Driver.Main2 (new string [] { $"-o={tmpDir}", "-c", warnerrAssembly.Location });
+					sw.Flush ();
+					ms.Position = 0;
+					var sr = new StreamReader (ms);
+					var output = sr.ReadToEnd ();
+
+					var arrCount = warnsToSearch.Length;
+					for (int i = 0; i < arrCount; i++) {
+						if (shouldFindWarn [i])
+							Assert.That (output, Does.Contain (warnsToSearch [i]), $"Did not find: {warnsToSearch [i]}");
+						else
+							Assert.That (output, Does.Not.Contain (warnsToSearch [i]), $"Did find: {warnsToSearch [i]}");
+					}
+				}
+			} finally {
+				Console.SetOut (stdout);
+				Console.SetError (stderr);
+			}
+		}
+
+		// We usually are in bin/debug|release
+		static readonly string XcodeFolderPath = $"{AppDomain.CurrentDomain.BaseDirectory}../../xcode";
+		static readonly string MonoMsBuildPath = "/Library/Frameworks/Mono.framework/Versions/Current/Commands/msbuild";
+		static readonly string MonoPath = "/Library/Frameworks/Mono.framework/Versions/Current/Commands/mono";
+		static readonly string ClangPath = "/usr/bin/clang";
+
+		[Test]
+		[TestCase ("NoInitInSubclassTest", "ConstructorsLib", "main.m", Platform.macOS, Configuration.Debug, "error: 'initWithId:' is unavailable")]
+		public void XcodeBuildErrorTest (string directoryTest, string csprojName, string objcmFileName, Platform platform, Configuration config, string errorToSearch)
+		{
+			Assert.IsTrue (Directory.Exists (XcodeFolderPath), "XcodeFolderPath");
+			Assert.IsTrue (File.Exists (MonoMsBuildPath), "MonoMsBuildPath");
+			Assert.IsTrue (File.Exists (MonoPath), "MonoPath");
+			Assert.IsTrue (File.Exists (ClangPath), "ClangPath");
+
+			var testcaseBaseDir = Path.Combine (XcodeFolderPath, directoryTest);
+			var tempWorkingDir = Xamarin.Cache.CreateTemporaryDirectory ();
+			var e4kOutputDir = Path.Combine (tempWorkingDir, "E4KOutput");
+
+			Asserts.RunProcess (MonoMsBuildPath, $"/p:Configuration={config} /p:IntermediateOutputPath={Path.Combine (tempWorkingDir, "obj")}/ /p:OutputPath={Path.Combine (tempWorkingDir, "DllOutput")} {Path.Combine (testcaseBaseDir, csprojName)}.csproj", "msbuildProc");
+
+			var eargs = new List<string> {
+					"-c",
+					$"{Path.Combine (tempWorkingDir, "DllOutput", csprojName)}.dll",
+					$"-o={e4kOutputDir}"
+				};
+			if (config == Configuration.Debug)
+				eargs.Add ("--debug");
+
+			Driver.Main2 (eargs.ToArray ());
+
+			// Sadly no C# 7 yet
+			// (string sdk, string arch, string sdkName, string minVersion) build_info;
```

Comment:
Could just make a struct with a "nice" constructor. I am looking for C#7 though.